api/v1alpha1/minicluster_types.go

```diff
@@ -56,6 +56,14 @@ type MiniClusterSpec struct {
 	// +optional
 	PullAlways bool `json:"pullAlways"`
 
+	// Allow the user to pull authenticated images
+	// By default no secret is selected. Setting
+	// this with the name of an already existing
+	// imagePullSecret will specify that secret
+	// in the pod spec.
+	// +optional
+	ImagePullSecrets []string `json:"imagePullSecrets"`
```

Comment:
I think for the time being we only have one container - and it's the same one so we can assume a common pull secret and just set this as a string.

If we wanted/needed to support multiple containers, e.g., I'm imagining that we have multiple images for some reason, and either it's the case that they have the same secret (and all need it) or some need a secret (others do not). I would create a Generic Secrets struct (used here) that takes the secret name, value, and then associated pod image. But since I don't think we have that case yet, let's reduce this to just be a string.

Reply:
We likely would want something like the above for additional environment variables to set, and I'm again OK waiting for an actual use case / request before implementing it.